### PR TITLE
[loki-distributed] Ingester Statefulset supports multiple PersistentVolume claims

### PR DESCRIPTION
**Pull request by chrisduong**
Hi,

I have a need to have 2 PVCs for Ingester Statefulset: one for _data_, and one for _wal_ files. However the template only allows one PVC.

I opened this PR for my suggestion.

### CODE REVIEW

**Comment by chrisduong** [comment]
Hi @zanhsieh, I don't know why Chart Testing install had failed, I checked the log it look kinda similar to the successful one. Could you tell me what is the reason or maybe try to re-run the test again.

File: charts/loki-distributed/values.yaml [hunk 1]
```diff
@@ -450,16 +450,19 @@ ingester:
...
-    # If set to "-", storageClassName: "", which disables dynamic provisioning.
-    # If empty or set to null, no storageClassName spec is
-    # set, choosing the default provisioner (gp2 on AWS, standard on GKE, AWS, and OpenStack).
-    storageClass: null
-    # -- Annotations for ingester PVCs
-    annotations: {}
+
+    claims:
+      # - name: data
+      #   size: 10Gi
```
Comment (zanhsieh) [inline charts/loki-distributed/values.yaml:456]:
```suggestion
      - name: data
        size: 10Gi
```

File: charts/loki-distributed/values.yaml [hunk 2]
```diff
@@ -450,16 +450,19 @@ ingester:
...
+      #   -- Storage class to be used.
+      #   If defined, storageClassName: <storageClass>.
+      #   If set to "-", storageClassName: "", which disables dynamic provisioning.
+      #   If empty or set to null, no storageClassName spec is
+      #   set, choosing the default provisioner (gp2 on AWS, standard on GKE, AWS, and OpenStack).
+      #   storageClass: null
```
Comment (zanhsieh) [inline charts/loki-distributed/values.yaml:462]:
```suggestion
      storageClass: null
```

File: charts/loki-distributed/README.md [hunk 3]
```diff
@@ -253,11 +253,9 @@ kubectl delete statefulset RELEASE_NAME-loki-distributed-querier -n LOKI_NAMESPA
 | ingester.livenessProbe | object | `{}` | liveness probe settings for ingester pods. If empty use `loki.livenessProbe` |
 | ingester.maxUnavailable | string | `nil` | Pod Disruption Budget maxUnavailable |
 | ingester.nodeSelector | object | `{}` | Node selector for ingester pods |
-| ingester.persistence.annotations | object | `{}` | Annotations for ingester PVCs |
+| ingester.persistence.claims | string | `nil` |  |
```
Comment (zanhsieh) [inline charts/loki-distributed/README.md:256]: `ingester.persistence.claims` type should be `list`.

File: charts/loki-distributed/values.yaml [hunk 4]
```diff
@@ -450,16 +450,19 @@ ingester:
...
-    size: 10Gi
-    # -- Storage class to be used.
-    # If defined, storageClassName: <storageClass>.
-    # If set to "-", storageClassName: "", which disables dynamic provisioning.
-    # If empty or set to null, no storageClassName spec is
-    # set, choosing the default provisioner (gp2 on AWS, standard on GKE, AWS, and OpenStack).
-    storageClass: null
-    # -- Annotations for ingester PVCs
-    annotations: {}
+
```
Comment (zanhsieh) [inline charts/loki-distributed/values.yaml:453]:
```suggestion
      # -- List of the ingester PVCs
      # @notationType -- list
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@chrisduong 
Please put the original pvc back.

File: charts/loki-distributed/README.md (hunk 3 above)
Comment (chrisduong) [reply charts/loki-distributed/README.md]: updated

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@chrisduong 
Please remove promtail from this PR. One PR can only contain one chart, otherwise CI will block merging.

**Review by zanhsieh — APPROVED** [review]

**Comment by bdols** [comment]
please consider at least bumping the major version on this chart for this kind of breaking change for those of us who are using statefulsets

**Pull request merged**